pkg/library/libraryinputresources/resource.go

```diff
@@ -85,9 +85,21 @@ func ResourceFromFile(location, fileTrimPrefix string) (*Resource, error) {
 	retFilename := strings.TrimPrefix(location, fileTrimPrefix)
 	retFilename = strings.TrimPrefix(retFilename, "/")
 
+	// If the file represents a list, derive the resource name from the file name.
```

Comment:
If this function will be only used to read must-gather files then based on [this](https://github.com/openshift/library-go/blob/573ff142f0a0c98340ce1ae2553499b61dcbcb70/pkg/manifestclient/list.go#L15) comment it should work.